src/main/java/org/generationcp/middleware/dao/SampleDao.java

```diff
@@ -384,6 +392,22 @@ public boolean hasSamples(final Integer studyId) {
 		return queryResults.isEmpty() ? false : true;
 	}
 
+	public boolean studyEntryHasSamples(final Integer studyId, final Integer entryId) {
+		final List queryResults;
+		try {
+			final SQLQuery query = this.getSession().createSQLQuery(SQL_STUDY_ENTRY_HAS_SAMPLES);
+			query.setParameter("studyId", studyId);
+			query.setParameter("entryId", entryId);
+			queryResults = query.list();
+
+		} catch (final HibernateException he) {
+			throw new MiddlewareException(
+					"Unexpected error in executing hasSamples(studyId = " + studyId + ") query: " + he.getMessage(),
```

Comment:
Fix the error message, hasSamples should be studyEntryHasSamples.

Reply:
Done